lms/models/lti_role.py

```diff
@@ -0,0 +1,147 @@
+from enum import Enum
+
+import sqlalchemy as sa
+from sqlalchemy.ext.hybrid import hybrid_property
+
+from lms.db import BASE
+
+
+class RoleType(str, Enum):
+    """Enum for the different types of role a user can have."""
+
+    INSTRUCTOR = "instructor"
+    LEARNER = "learner"
+    ADMIN = "admin"
+
+    @classmethod
+    def parse_lti_role(cls, role):
+        """Parse an LTI role string into one of our role types."""
+
+        # We have to do this work around because Enums can't have private
+        # attributes. They are just interpreted as extra values for the enum.
+        return _RoleParser.parse_role(role)
+
+
+class LTIRole(BASE):
+    """Model for LTI role strings and our interpretation of them."""
+
+    __tablename__ = "lti_role"
+
+    # Export for others convenience
+    Type = RoleType
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    _value = sa.Column("value", sa.UnicodeText(), nullable=False, unique=True)
+    """The raw string from LTI params."""
+
+    type = sa.Column("type", sa.Enum(RoleType), nullable=False, unique=False)
+    """Our interpretation of the value."""
+
+    @hybrid_property
+    def value(self):
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        self._value = value
+        self.type = RoleType.parse_lti_role(value)
+
+
+class _RoleParser:
+    """Close collaborator class for parsing roles."""
+
+    @classmethod
+    def parse_role(cls, role) -> RoleType:
+        return (
+            cls._parse_v13_role(role) or cls._parse_v11_role(role) or RoleType.LEARNER
+        )
+
+    _V11_INSTRUCTOR_STRINGS = (
+        "instructor",
+        "teachingassistant",
+        "contentdeveloper",
+        "faculty",
+        "mentor",
+        "staff",
```

Comment:
This is **not** the same list as we use in other places. We count more things here to be instructors.

This probably needs to be vetted by some outside folk?

In any case currently this does not interact with the existing role system at all and shouldn't cause any trouble. We may want to unify them in the future though.